src/Message/AbstractRequest.php

```diff
@@ -30,26 +28,15 @@ public function getTransactionId()
 
     protected function sendRequest($method, $endpoint, $data = null)
     {
-        $this->httpClient->getEventDispatcher()->addListener('request.error', function (Event $event) {
-            /**
-             * @var \Guzzle\Http\Message\Response $response
-             */
-            $response = $event['response'];
-
-            if ($response->isError()) {
-                $event->stopPropagation();
-            }
-        });
-
         $httpRequest = $this->httpClient->createRequest(
             $method,
             $this->endpoint . $endpoint,
             array(
                 'Authorization' => 'Bearer ' . $this->getApiKey()
             ),
-            $data
+            json_encode($data)
```

Comment:
Untill now $data was an array, I found an example in ruudk/payment-mollie-bundle. which is using:
$response = $this->gateway->completePurchase(array(
     'transactionReference' => $trackingId
))->send();

If I use just `$data`, I will get the following error: `Invalid resource type: array` (in GuzzleHttp\Psr7\stream_for). So it has to be parsed to a string. Input (array) stays the same.